### DISCUSSION TITLE
Some types are not fully retrieved when placed on the same package

### DISCUSSION THREAD

**Discussion by glureau** [LemonAppDev/konsist#1344] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.16.1

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm checking the parameter type of a class which is roughly similar to

Foo.kt
```kotlin
package a.b
data class Foo(val status: FooStatus = FooStatus.OK)
```

FooStatus.kt
```kotlin
package a.b
enum class FooStatus { OK, KO }
```

When I do `KoParameterDeclaration.type.asClassDeclaration()` I get a null. But if I import explicitly FooStatus then I get a proper class declaration.

As a byproduct of this bug I guess, the `KoParameterDeclaration.type.packagee.containingFile` is wrong as it's showing the Foo.kt file (it should be FooStatus.kt). And a call to `KoParameterDeclaration.type.declaration` is returning a KoExternalDeclarationCore, but this enum class is not really external indeed, should be a KoClassDeclarationCore I guess.

### Actual behaviour

When I do `KoParameterDeclaration.type.asClassDeclaration()` I get a null.

### Expected behaviour

When I do `KoParameterDeclaration.type.asClassDeclaration()` I should have the KoClassDeclaration of FooStatus.

**Comment by nataliapeterwas** [comment, marked as answer]
Thank you for reporting the problem.
We will fix this issue in the next release: [PR](https://github.com/LemonAppDev/konsist/pull/1353)

**Comment by nataliapeterwas** [comment]
Konsist `0.17.0` has been released! This release contains a fix for this issue :) See full release log: [Release log](https://github.com/LemonAppDev/konsist/releases/tag/v0.17.0)
